x/pubkey/keeper/keeper.go

```diff
@@ -88,3 +94,94 @@ func (k Keeper) GetValidatorKeys(ctx context.Context, validatorAddr string) (res
 	}
 	return result, nil
 }
+
+// GetAllValidatorPubKeys returns all validator public keys in the store.
+func (k Keeper) GetAllValidatorPubKeys(ctx context.Context) ([]types.ValidatorPubKeys, error) {
+	var valPubKeys []types.ValidatorPubKeys
+
+	itr, err := k.pubKeys.Iterate(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer itr.Close()
+
+	var currentVal []byte
+	for ; itr.Valid(); itr.Next() {
+		kv, err := itr.KeyValue()
+		if err != nil {
+			return nil, err
+		}
+
+		// Skip if the validator has already been processed.
+		if bytes.Equal(kv.Key.K1(), currentVal) {
+			continue
+		}
```

Comment:
~~Why do we need this check?~~

Ooh, I guess the iterator is over the entire store, so with multiple proving schemes we would get multiple hits for the same validator. And since the iterator is 'sorted' we can simply look at the previous iterations value?

Reply:
Yes exactly. I suppose I could add a prefix iterator inside the for loop. Will look into this tmrw.

Reply:
Hm, It doesn't look like there is a better way to iterate over K1s.